prime-dotnet-webapi/Services/EnrolleeAgreementService.cs

```diff
@@ -124,6 +130,16 @@ public async Task<Agreement> GetCurrentAgreementAsync(int enrolleeId)
                 .FirstAsync(at => at.EnrolleeId == enrolleeId);
         }
 
+        public async Task<AgreementType> GetCurrentAgreementTypeAsync(int enrolleeId)
+        {
+            return await _context.Agreements
+                .Include(a => a.AgreementVersion)
```

Comment:
You shouldn't need to `.Include()` the versions if you later do a `.Select()` to get them